File: app/src/main/java/com/example/waveplayer/SongDAO.java
package com.example.waveplayer;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface SongDAO {

    @Insert
    void insertAll(Song... songs);

    @Delete
    void delete(Song user);

    @Query("SELECT * FROM songs WHERE id = :song_id")
    Song getSong(Long song_id);

    @Query("SELECT * FROM songs")
    List<Song> getAll();

}

File: app/src/main/java/com/example/waveplayer/random_playlist/SongDAO.kt
package com.example.waveplayer.random_playlist

import androidx.room.*

@Dao
interface SongDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(vararg songs: Song)
    @Delete
    fun delete(user: Song)
    @Query("SELECT * FROM songs WHERE id = :song_id")
    fun getSong(song_id: Long): Song
    @Query("SELECT * FROM songs")
    fun getAll(): MutableList<Song>
    @Query("DELETE FROM songs")
    fun deleteAll()
}